transport/src/main/java/io/el/channel/ChannelEventLoopGroup.java

```diff
@@ -6,11 +6,16 @@ public interface ChannelEventLoopGroup extends EventLoopGroup {
 
   /**
    * Returns {@link ChannelEventLoop} it manages
-   * */
+   */
   ChannelEventLoop next();
 
   /**
    * Bind {@link Channel} with the {@link ChannelEventLoop} one of it manages
-   * */
+   */
   ChannelPromise register(Channel channel);
+
+  /**
+   * Register using {@link ChannelPromise} with the {@link ChannelEventLoop} one of it manages
+   */
+  ChannelPromise register(ChannelPromise promise);
```

Comment:
How about restricting the way to register the channel to the event loop? I think for now it's enough to only having `ChannelPromise register(Channel channel)` interface. 

`ChannelPromise register(ChannelPromise promise)` method you added to ChannelSingleThreadEventLoop should be internal